app/assets/v2/js/board.js

```diff
@@ -423,7 +468,71 @@ Vue.mixin({
         handleError(err);
       });
     },
+    checkNetwork() {
+      return 'rinkeby';
+      const supportedNetworks = ['rinkeby', 'mainnet'];
+
+      if (!supportedNetworks.includes(document.web3network)) {
+        _alert('Unsupported network', 'error');
+        throw new Error('Please connect a wallet');
+      }
+      return document.web3network;
+    },
+    async complete(redemptionId, tokenAmount) {
+      const vm = this;
+
+      try {
+        const network = vm.checkNetwork();
+        const amount = web3.utils.toWei(String(tokenAmount));
+        const [user] = await web3.eth.getAccounts();
+
+        indicateMetamaskPopup();
+        const pToken = await new web3.eth.Contract(
+          document.contxt.ptoken_abi,
+          document.current_ptoken_address
```

Comment:
`document.current_ptoken_address` is undefined, therefore this function fails. Since we are clicking a button from the dashboard, we need to get the pToken address from the `redemptionId` input, since it's specific to the dashboard button clicked

Reply:
I added the tokenAddress as parameter